### PR TITLE
Retry rolling upgrade junit tests

### LINKED ISSUE #97200: Add index version numbers to cluster upgrade tests
BwC & upgrade cluster tests have the node versions to use passed in as environment variables (eg https://github.com/elastic/elasticsearch/blob/main/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/AbstractRollingTestCase.java#L32, https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/rolling-upgrade-multi-cluster/src/test/java/org/elasticsearch/upgrades/AbstractMultiClusterUpgradeTestCase.java#L69). After we migrate to the IndexVersion class, there will not be a mapping between Version and IndexVersion. This means we will need to add index version to the test setup & environment variables in various tests.

How will we map between the node version & corresponding indexversion? Or does that not particularly matter once they're separated?

### PR DESCRIPTION
**Pull request by thecoop**
Retry the rolling upgrade junit tests from https://github.com/elastic/elasticsearch/pull/99572, reverted by #99750

This re-closes #97200

### CODE REVIEW

File: qa/full-cluster-restart/build.gradle [hunk 1]
```diff
@@ -19,3 +19,8 @@ BuildParams.bwcVersions.withIndexCompatible { bwcVersion, baseName ->
     systemProperty("tests.old_cluster_version", bwcVersion)
   }
 }
+
+tasks.withType(Test).configureEach {
+  // CI doesn't like it when there's multiple clusters running at once
+  maxParallelForks = 1
```
[the remark this answers is not in the document]
Comment (thecoop) [reply qa/full-cluster-restart/build.gradle:25]: It's not the tests themselves, its that the CI clusters can't cope with the load and so the clusters slow down so much the tests time out

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/bwc

File: qa/full-cluster-restart/build.gradle [hunk 2]
```diff
@@ -19,3 +19,8 @@ BuildParams.bwcVersions.withIndexCompatible { bwcVersion, baseName ->
     systemProperty("tests.old_cluster_version", bwcVersion)
   }
 }
+
+tasks.withType(Test).configureEach {
```
Comment (thecoop) [inline qa/full-cluster-restart/build.gradle:23]: @breskeby will this turn off parallelism for all test tasks everywhere, or just test tasks in this module?

Comment (breskeby) [reply qa/full-cluster-restart/build.gradle:23]: just all test tasks within this `full-cluster-restart` module

**Comment by thecoop** [comment]
This looks much happier - bwc tests have passed with no issues

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/bwc

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/bwc

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/RestrictedBuildApiService.java [hunk 3]
```diff
@@ -67,6 +67,7 @@ private static ListMultimap<Class<?>, String> createLegacyRestTestBasePluginUsag
         map.put(LegacyRestTestBasePlugin.class, ":qa:remote-clusters");
         map.put(LegacyRestTestBasePlugin.class, ":qa:repository-multi-version");
         map.put(LegacyRestTestBasePlugin.class, ":qa:rolling-upgrade");
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/RestrictedBuildApiService.java:69]: we should be able to remove `:qa:rolling-upgrade` from this list if the project doesn't use any legacy plugins anymore

**Review by breskeby — APPROVED** [review]
minor change. otherwise lgtm

File: qa/rolling-upgrade/build.gradle [hunk 4]
```diff
@@ -6,135 +6,30 @@
...
-      excludeList.addAll(["upgraded_cluster/30_vector_search/*"])
-    } else if (bwcVersion.before("8.6.0")) {
-      excludeList.addAll(["upgraded_cluster/30_vector_search/Search byte indices created in old cluster"])
-    }
-    if (excludeList.isEmpty() == false) {
-      systemProperty 'tests.rest.blacklist', excludeList.join(',')
-    }
-  }
+testClusters.configureEach {
+  setting 'xpack.security.enabled', 'false'
```
Comment (rjernst) [inline qa/rolling-upgrade/build.gradle:29]: Isn't this set in the test itself now? It seems odd the new style testclusters even has the ability to set settings on each cluster, these should always be in the .java files.

**Review by rjernst — APPROVED** [review]
LGTM

File: qa/rolling-upgrade/build.gradle (hunk 4 above)
Comment (thecoop) [reply qa/rolling-upgrade/build.gradle]: Er, yes. I added this because some tests were failing to use HTTPS properly. This is probably no longer the case

**Pull request merged**